packages/dds/sequence/src/intervalIndex/index.ts

```diff
@@ -0,0 +1,8 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+export { SequenceIntervalIndexes } from "./SequenceIntervalIndexes";
+export { createOverlapping } from "./overlappingSequenceIntervalsIndex";
```

Comment:
@Abe27342  I have a concern at this point. Although I have enclosed the `createOverlapping` function within the `sequenceIntervalIndexFactory`, it appears that the API-extractor is still requiring it to be publicly accessible. Is there any possible means to avoid this redundant exporting?

Reply:
The api-extractor complaint is because it can't resolve the type of `createOverlapping` with your current structure. You could fix it by doing this:

```typescript
/*!
 * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
 * Licensed under the MIT License.
 */

import { Client } from "@fluidframework/merge-tree";
import { SequenceIntervalIndexes } from "./SequenceIntervalIndexes";
import { createOverlapping } from "./overlappingSequenceIntervalsIndex";

export interface SequenceIntervalIndexFactories {
	createOverlappingIndex(client: Client): SequenceIntervalIndexes.Overlapping;
}

/**
 * The factory functions for creating sequence interval indexes
 */
export const sequenceIntervalIndexFactory: SequenceIntervalIndexFactories = {
	createOverlappingIndex: createOverlapping,
};
```

but on second thought, I'm not even sure this pattern tree-shakes nicely. I think it's probably better to just export factory functions like `createOverlappingSequenceIntervalIndex` etc.

I still like the namespace organization for the types, though.